hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/TransactionHandler.java

```diff
@@ -16,30 +16,40 @@
 
 package com.hedera.node.app.spi.workflows;
 
-import com.hedera.node.app.spi.records.RecordBuilder;
+import com.hedera.hapi.node.transaction.TransactionBody;
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 /**
  * A {@code TransactionHandler} contains all methods for the different stages of a single operation.
  */
 public interface TransactionHandler {
 
+    /**
+     * Validates a {@link TransactionBody}
+     *
+     * @param txBody the {@link TransactionBody} to validate
+     * @throws NullPointerException if {@code txBody} is {@code null}
+     * @throws PreCheckException if the transaction is invalid
+     */
+    default void validate(TransactionBody txBody) throws PreCheckException {
```

Comment:
I think this is the same as the "pure check" method Neeha added?

Reply:
Right, at some point, we called it `validate()`. I will pull the latest changes and clean this up.

Reply:
Fixed